sql: anonymize schema names, owner names, and enum values

solongordon opened the pull request:
Schema and owner names were improperly being represented by strings
rather than `tree.Name`s in the AST. This meant they would not be
properly anonymized in telemetry.

ENUM values had the same problem, but we cannot use `tree.Name` in
this case because we need the values to be quoted. I created a custom AST
node for this case.

Fixes #55385

Release note: None

solongordon committed:
sql: anonymize schema and owner names

Schema and owner names were improperly being represented by strings
rather than `tree.Name`s in the AST. This meant they would not be
properly anonymized in telemetry.

Fixes #55385

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55473)
<!-- Reviewable:end -->

solongordon commented:
On further inspection I think ENUMs are not properly anonymized either. I'll circle back to fix that.

lucy-zhang commented:
I see that https://github.com/cockroachdb/cockroach/pull/55398 deals with owner names by using the newly introduced `security.SQLUsername` type (and I thought there were plans underway to backport part of it). Is using `tree.Name` an intermediate stage to fix the immediate problem?

knz reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> This fixes the case of schema and role names very nicely. Thanks for the quick turnaround :100: 

For enum values unfortunately we can't use tree.Name, because it doesn't quote properly upon pretty-printing.
However we can define a custom AST node type and add a `Format()` method that does the anonymization for them.

> Reviewed 20 of 20 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55473#-:-MJW7cC0-mS2ho2H831k:b5lz53)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @lucy-zhang)



<!-- Sent from Reviewable.io -->

solongordon committed:
sql: anonymize enum values

ENUM values were improperly being represented as strings in the AST,
which meant that they were not anonymized in telemetry. I created a new
AST node type for them which respects anonymization.

Fixes #55385

Release note: None

solongordon commented:
@knz I added a second commit to handle enum values. Would you please take another look?

@lucy-zhang I wasn't sure whether #55398 would be backported so I wanted to put in a simple fix for owner names. When that is merged it can replace my owner changes here.

thoszhang reviewed (commented):
Changes LGTM. Should this have formatting tests?

> Reviewed 19 of 20 files at r1, 1 of 1 files at r2, 8 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55473#-:-MJXIi2R1IvHDBQjfpd3:b-fk1u02)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

solongordon commented:
Simple formatting has pre-existing tests in parse_test.go, but let me augment with a few tests for anonymized formatting.

solongordon commented:
Added anonymization tests.

thoszhang reviewed (approved):
> Reviewed 9 of 9 files at r4, 9 of 9 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55473)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

solongordon commented:
bors r+

knz reviewed (commented):
> Reviewed 1 of 1 files at r2, 9 of 9 files at r7.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55473)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->
